### PR TITLE
fix closing editor panes

### LINKED ISSUE #55864: Saving your code closes editor panes
### Describe the Issue

I didn't test this too thoroughly, but when writing code for one of the savable projects, I saved my code and the CSS pane closed.

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/build-a-survey-form-project/build-a-survey-form

### Steps to Reproduce

1. Go to https://www.freecodecamp.org/learn/2022/responsive-web-design/build-a-survey-form-project/build-a-survey-form
2. Open both the HTML and CSS editor panes
3. Write some code in one of them
4. Click the "save your code" button or press `ctrl+s`
5. The CSS pane closes

### Expected behavior

The panes should stay how I had them.

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

Possibly somewhat related: https://github.com/freeCodeCamp/freeCodeCamp/issues/55619

### PR DESCRIPTION
**Pull request by Inventoris**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #55864

Hi! I looked into the issue and noticed that when saving the code, createFiles is executed, in which visibleEditors is set to one main tab by default. I added code so that if state.visibleEditors contains current tabs at the time createFiles is launched, then visibleEditors will contain exactly them.

### CODE REVIEW

File: client/src/templates/Challenges/redux/index.js [hunk 1]
```diff
@@ -79,11 +79,19 @@ export const reducer = handleActions(
...
-      ...state,
-      challengeFiles: payload,
-      visibleEditors: { [getTargetEditor(payload)]: true }
-    }),
+    [actionTypes.createFiles]: (state, { payload }) => {
+      let visibleEditors = state.visibleEditors;
+
+      if (Object.keys(visibleEditors).length === 0) {
+        visibleEditors = { [getTargetEditor(payload)]: true };
+      };
```
Comment (gikf) [inline client/src/templates/Challenges/redux/index.js:87]:
```suggestion
      }
```
This semicolon is cause of the CI error.

**Review by Sembauke — APPROVED** [review]
Tested locally, works like a charm.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Hi @Inventoris, thanks for taking the time to fix this. However, I think the root of the problem is that the `createFiles` action has too many jobs - it should put the challenge files in the redux store and nothing else.

If we do that, and update the ui via another action (`initVisibleEditors`, say), there's no need for any conditionals in the reducer and no chance of weird interactions.

**Review by ojeytonwilliams — APPROVED** [review]
Perfect. Thanks, @Inventoris

LGTM :+1:

**Pull request merged**